### PR TITLE
Mudensity to metersetmap

### PR DESCRIPTION
**Pull request by SimonBiggs**
So, I was just laying in bed awake... couldn't sleep. Figured I'd 'count sheep'... instead I counted Ctrl + F replacements :).

### CODE REVIEW

File: lib/pymedphys/__init__.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 """Module docstring."""
 
-from . import dicom, electronfactors, mosaiq, mudensity, trf, wlutz
+from . import dicom, electronfactors, mosaiq, metersetmap, mudensity, trf, wlutz
```
Comment (sjswerdloff) [inline lib/pymedphys/__init__.py:3]: mudensity should no longer be imported, shouldn't mudensity get removed?

File: lib/pymedphys/_dev/docs.py [hunk 2]
```diff
@@ -9,6 +9,12 @@
 
 
 def build_docs(args):
+    subprocess.check_call(
```
Comment (sjswerdloff) [inline lib/pymedphys/_dev/docs.py:12]: this should have been done in a different PR.
Especially when a PR has many files involved, the PR should have a single intent.

File: lib/pymedphys/_experimental/paulking/devicesfilm.py [hunk 3]
```diff
@@ -24,7 +24,7 @@
 
 
 def read_narrow_png(file_name, step_size=0.1):
-    """  Extract a an relative-density profilee from a narrow png file.
+    """  Extract a an relative-density profile from a narrow png file.
```
Comment (sjswerdloff) [inline lib/pymedphys/_experimental/paulking/devicesfilm.py:27]:
```suggestion
    """  Extract a relative-density profile from a narrow png file.

```

File: lib/pymedphys/__init__.py (hunk 1 above)
Comment (SimonBiggs) [reply lib/pymedphys/__init__.py:3]: For backwards compatibility. `pymedphys.mudensity.calculate` was part of our API.

See the new `mudensity.py` that this is importing from:

https://github.com/pymedphys/pymedphys/blob/60c4cbda800750022a0d5817c08e30670ce714a1/lib/pymedphys/mudensity.py#L1-L23

File: lib/pymedphys/_dev/docs.py (hunk 2 above)
Comment (SimonBiggs) [reply lib/pymedphys/_dev/docs.py:12]: Yup, good call. See https://github.com/pymedphys/pymedphys/pull/1169.

If that gets merged first then it will no longer be a part of the changelog in this PR.

File: lib/pymedphys/__init__.py (hunk 1 above)
Comment (SimonBiggs) [reply lib/pymedphys/__init__.py:3]: > If that's the case, then maybe this could have a comment indicating that it is being deprecated and should be removed by version 0.xy ?

Yup, deprecation notice already included. I haven't specified a release in which it'll be removed. I suspect we can have a "last notice" release down the line where all deprecation notices get a "this will be removed in the next version" addition. And then in the following version each deprecation gets removed.

File: lib/pymedphys/docs/conf.py [hunk 4]
```diff
@@ -15,12 +15,8 @@
 
 # pylint: disable = C0413, C0103
 
-import os
-
 import pymedphys
 
-os.system("pandoc CHANGELOG.md --from markdown --to rst -s -o release-notes.rst")
```
Comment (sjswerdloff) [inline lib/pymedphys/docs/conf.py:22]: paired with introduction of this as check_process above in this PR.
the pair probably should have been done in a separate PR

Comment (SimonBiggs) [reply lib/pymedphys/docs/conf.py:22]: Yup see https://github.com/pymedphys/pymedphys/pull/1169 I cherry-picked that commit into a separate PR.

File: lib/pymedphys/mudensity.py [hunk 5]
```diff
@@ -1,30 +1,23 @@
-"""A suite of functions to determine the MU Density for a range of
-formats
+from ._vendor.deprecated import deprecated as _deprecated
```
Comment (sjswerdloff) [inline lib/pymedphys/mudensity.py:1]: this answers my question earlier about import of this and deprecation.
Is it feasible to indicate at what version this will be obsolete as a vigorous warning to the user/programmer to change their code?

Comment (SimonBiggs) [reply lib/pymedphys/mudensity.py:1]: See https://github.com/pymedphys/pymedphys/pull/1168#discussion_r520927280, let me know what you think.

File: lib/pymedphys/tests/metersetmap/test_metersetmap_regression.py [hunk 6]
```diff
@@ -28,16 +28,18 @@
...
     regress_test_arrays = np.load(data_filepath)
 
     mu = regress_test_arrays["mu"]
     mlc = regress_test_arrays["mlc"]
     jaw = regress_test_arrays["jaw"]
 
-    cached_mu_density = regress_test_arrays["mu_density"]
-    mu_density = pymedphys.mudensity.calculate(mu, mlc, jaw)
-    assert np.allclose(mu_density, cached_mu_density, atol=0.1)
+    # MetersetMap used to be called mu_density. The regression test data
```
Comment (sjswerdloff) [inline lib/pymedphys/tests/metersetmap/test_metersetmap_regression.py:41]: so is this a TODO and the regression test data will be renamed?

**Review by sjswerdloff — APPROVED** [review]
but please review comments

File: lib/pymedphys/tests/metersetmap/test_metersetmap_regression.py (hunk 6 above)
Comment (SimonBiggs) [reply lib/pymedphys/tests/metersetmap/test_metersetmap_regression.py:41]: I wasn't considering changing it. I tend to be adverse to changing regression testing.

**Comment by SimonBiggs** [comment]
> but please review comments

Cheers :),

Let me know your thoughts.

Also, if you use the GitHub "start a review" functionality then all of the comments come through in one batch, and I can then do the same in reply, essentially limiting the number of notifications landing in each our email inbox to one per review:

image

**Comment by sjswerdloff** [comment]
> > but please review comments
> 
> @sjswerdloff comments reviewed :), let me know your if you're happy for me to merge re the responses.
> 
> Also, if you use the GitHub "start a review" functionality then all of the comments come through in one batch, and I can then do the same in reply, essentially limiting the number of notifications landing in each our email inbox to one per review:
> 
> image

Ah... will do, thank you.
And... happy for the merge.  Satisfied with all responses.

**Comment by SimonBiggs** [comment]
Cheers :),

I'll pause this merge until https://github.com/pymedphys/pymedphys/pull/1169 lands. I'll merge after that though.
